### PR TITLE
FC-934 Update/nodejs with functions for time-travel, auth, ledger info, ...

### PR DESCRIPTION
**Pull request by ldw1007**
The following updates for Node.js library:

* flureenjs.clj - new addition; fix issue where node.js could not find dbfunctions in bindings
* flureenjs.cljs - streamlined code, made some changes for sonarqube, ADDED functions
* connection_js.cljs - needed to add token to connection for storage requests.  We need a better way to do this -> will be tracked under FC-902.
* dbfunctions - extended support for node.js (e.g., find-fn)

### CODE REVIEW

File: src/fluree/db/dbfunctions/core.cljc [hunk 1]
```diff
@@ -1,5 +1,6 @@
 (ns fluree.db.dbfunctions.core
-  (:require [clojure.tools.reader.edn :as edn]
+  (:require #?(:clj  [clojure.tools.reader.edn :as edn])
```
Comment (zonotope) [inline src/fluree/db/dbfunctions/core.cljc:2]: if you `:refer [read-string]` here, then you would be able to get rid of the reader conditional below where you use this function in both platforms

File: src/fluree/db/dbfunctions/core.cljc [hunk 2]
```diff
@@ -30,6 +31,11 @@
   #?(:clj  (reset! db-fn-cache (cache/fifo-cache-factory {} :threshold 500))
      :cljs (reset! db-fn-cache (cache/lru-cache-factory {} :threshold 500))))
 
+(defn- read-str
+  "Like read-string but abstracts CLJ / CLJS differences"
+  [s]
+  #?(:clj  (edn/read-string s)
```
Comment (zonotope) [inline src/fluree/db/dbfunctions/core.cljc:37]: This is the reader conditional you can eliminate if you `:refer [read-string]` on both platforms.

File: src/fluree/db/dbfunctions/core.cljc [hunk 3]
```diff
@@ -111,12 +117,46 @@
...
+                                                     :from      ["_fn/name" (name fn-name)]}
+                                        res*        (<? (dbproto/-query db query))
+                                        _           (if (empty? res*)
+                                                      (throw (ex-info (str "Unknown function: " (pr-str fn-name))
+                                                                      {:status 400
+                                                                       :error  :db/invalid-fn})))
+                                        params      (read-str (get res* "_fn/params"))
+                                        code        (<? (resolve-fn db (read-str (get res* "_fn/code")) funType params))
+                                        spec        (get res* "_fn/spec")
+                                        params'     (mapv (fn [x] (symbol x)) params)
```
Comment (zonotope) [inline src/fluree/db/dbfunctions/core.cljc:147]: minor, but you can combine these to lines like this

```clojure
params' (->> params
             (mapv (fn [x] (symbol x)))
             (cons '?ctx)
             (into []))
```
again, this is minor and subject to tastes, so I'm only presenting it here as a suggestion. Take it or leave it.

**Review by zonotope — APPROVED** [review]
👍🏾

File: src/fluree/db/dbfunctions/core.cljc (hunk 1 above)
[the remark this answers is not in the document]
Comment (ldw1007) [reply src/fluree/db/dbfunctions/core.cljc]: excluded clojure.core read-string & consolidated read-string logic.

File: src/fluree/db/dbfunctions/core.cljc (hunk 3 above)
Comment (ldw1007) [reply src/fluree/db/dbfunctions/core.cljc]: made the suggested change.  :) Ran tests to validate

File: src/fluree/db/dbfunctions/core.cljc (hunk 2 above)
Comment (ldw1007) [reply src/fluree/db/dbfunctions/core.cljc]: excluded clojure.core read-string & consolidated read-string logic.

File: src/fluree/db/dbfunctions/core.cljc (hunk 1 above)
Comment (ldw1007) [reply src/fluree/db/dbfunctions/core.cljc]: excluded clojure.core read-string & consolidated read-string logic.

**Comment by ldw1007** [comment]
Made requested/suggested changes.
Will resolve conversation & merge.

**Pull request merged**